### DISCUSSION TITLE
Setting / editing / changing / manipulating local bash variables inside fzf execute()

### DISCUSSION THREAD

**Discussion by MikeLemo1** [junegunn/fzf#3669] [Q&A]
I'm basically trying to make a somewhat of a file explorer based of fzf that with single fzf execution I'll be able to reload fd entry and toggle between files or directories and in different directory entries and in order to do that I'll need some way of declaring variables inside fzf execution (Through the command opts) but I'm struglling with changing/setting a variable to be able to then read off of it and toggle / use it as an entry in the `fd` query
Also but I't quite related I'm trying to use the fzf Prompt as a variable and query toggler but since my working across a 2D matrix option  I need finer read control access to each word(divided by space " ") and access it similar to `${FZF_PROMPT[2]}` to say access the second or third word it the ${FZF_PROMPT} but first you need to declare it into another array like; `MY_ARR=($FZF_PROMPT)` which I'm not sure how to do and tryed with `execute()` with no luck.
Any idea?

I'm also trying to execute internal function say `myFunc() { echo "Test from the other side successful" }`
but not sure how to...

**Comment by MikeLemo1** [comment]
Dumb question... 
Just have to use `;`  between var declaraction and a command e.g.:
* Prints the third element of the `$FZF_QUERY` after it's been split and "arrayed" into `MY_ARR`
```bash
  --bind "ctrl-r:execute: MY_ARR=($FZF_QUERY) ; echo ${MY_ARR[2]}"
```
where I instead tryed using  `|`, `||`, `&`, `&&`
Closing.

**Reply by chzurb** [reply]
@MikeLemo1 Would you be willing to share a bit of your code.  I think I'm trying to do the same thing as you but I want to be able to BOTH 

1. reload a new src using the `reload` action (like you i'm switching between files and dirs )
2. AND SIMULTANEOUSLY change "state" of certain variables to "follow" the different reload "contexts"
3. while staying in fzf single instance

In essence I want to push the VARS up to the same process that FZF is running in and NOT necessarily just down some `execute` action which is a separate subprocess.  Simply because i need to have the variables available *between* the different reloads and not just *isolated* down each `execute` action if that makes any sense.

thanks
